DOC: please improve CONTRIBUTING.md with minimal build instructions

I do acknowledge that I have no clue (yet) in go, but I do not consider my actions unreasonable:
```shell
$> git clone https://github.com/GoogleCloudPlatform/container-diff                                           
Cloning into 'container-diff'...                                   
remote: Counting objects: 3253, done.
remote: Compressing objects: 100% (26/26), done.
remote: Total 3253 (delta 13), reused 23 (delta 8), pack-reused 3219
Receiving objects: 100% (3253/3253), 6.06 MiB | 5.46 MiB/s, done.
Resolving deltas: 100% (1551/1551), done.

$> cd container-diff 

$> make
can't load package: package github.com/GoogleCloudPlatform/container-diff: cannot find package "github.com/GoogleCloudPlatform/container-diff" in any of:
	/usr/lib/go-1.9/src/github.com/GoogleCloudPlatform/container-diff (from $GOROOT)
	/home/yoh/gocode/src/github.com/GoogleCloudPlatform/container-diff (from $GOPATH)
mkdir -p ./out
GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -tags "container_image_ostree_stub containers_image_openpgp" -ldflags "-X github.com/GoogleCloudPlatform/container-diff/version.version=v0.5.2" -o out/container-diff-linux-amd64 github.com/GoogleCloudPlatform/container-diff
can't load package: package github.com/GoogleCloudPlatform/container-diff: cannot find package "github.com/GoogleCloudPlatform/container-diff" in any of:
	/usr/lib/go-1.9/src/github.com/GoogleCloudPlatform/container-diff (from $GOROOT)
	/home/yoh/gocode/src/github.com/GoogleCloudPlatform/container-diff (from $GOPATH)
Makefile:45: recipe for target 'out/container-diff-linux-amd64' failed
make: *** [out/container-diff-linux-amd64] Error 1
```
so clearly it expects some "setup" before being able to be built.  Any documented guidance would be of help I guess

cool, so might be a nice (despite probably being too obvious for go experts) addition to CONTRIBUTING.  Overall it would be nice to have a section on how to establish development env, so it doesn't pollute system/user-wide `go` environment.  Here is my humble attempt revealing my ignorance on the subject:
```shell
yoh@hopa:/tmp> mkdir /tmp/GOPATH

yoh@hopa:/tmp> export GOPATH=/tmp/GOPATH

yoh@hopa:/tmp> git clone https://github.com/GoogleCloudPlatform/container-diff $GOPATH/src/github.com/GoogleCloudPlatform/container-diff
Cloning into '/tmp/GOPATH/src/github.com/GoogleCloudPlatform/container-diff'...
remote: Counting objects: 3253, done.
remote: Compressing objects: 100% (26/26), done.
remote: Total 3253 (delta 13), reused 23 (delta 8), pack-reused 3219
Receiving objects: 100% (3253/3253), 6.06 MiB | 4.67 MiB/s, done.
Resolving deltas: 100% (1551/1551), done.

yoh@hopa:/tmp> cd $GOPATH/src/github.com/GoogleCloudPlatform/container-diff

yoh@hopa:/tmp/GOPATH/src/github.com/GoogleCloudPlatform/container-diff> make
go: cannot find GOROOT directory: /home/yoh/deb/perspect/container-diff/gocode
go: cannot find GOROOT directory: /home/yoh/deb/perspect/container-diff/gocode
go: cannot find GOROOT directory: /home/yoh/deb/perspect/container-diff/gocode
make: *** No rule to make target 'out/container-diff--amd64', needed by 'out/container-diff'.  Stop.

yoh@hopa:/tmp/GOPATH/src/github.com/GoogleCloudPlatform/container-diff> cd -
/tmp

yoh@hopa:/tmp> mkdir GOROOT

*yoh@hopa:/tmp> export GOROOT=/tmp/GOROOT

yoh@hopa:/tmp> cd -
/tmp/GOPATH/src/github.com/GoogleCloudPlatform/container-diff

yoh@hopa:/tmp/GOPATH/src/github.com/GoogleCloudPlatform/container-diff> make
mkdir -p ./out
GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -tags "container_image_ostree_stub containers_image_openpgp" -ldflags "-X github.com/GoogleCloudPlatform/container-diff/version.version=v0.5.2" -o out/container-diff-linux-amd64 github.com/GoogleCloudPlatform/container-diff
pkg/util/docker_utils.go:20:2: cannot find package "archive/tar" in any of:
	/tmp/GOPATH/src/github.com/GoogleCloudPlatform/container-diff/vendor/archive/tar (vendor tree)
	/tmp/GOROOT/src/archive/tar (from $GOROOT)
	/tmp/GOPATH/src/archive/tar (from $GOPATH)
differs/apt_diff.go:20:2: cannot find package "bufio" in any of:
	/tmp/GOPATH/src/github.com/GoogleCloudPlatform/container-diff/vendor/bufio (vendor tree)
	/tmp/GOROOT/src/bufio (from $GOROOT)
	/tmp/GOPATH/src/bufio (from $GOPATH)
pkg/util/docker_utils.go:21:2: cannot find package "bytes" in any of:
	/tmp/GOPATH/src/github.com/GoogleCloudPlatform/container-diff/vendor/bytes (vendor tree)
	/tmp/GOROOT/src/bytes (from $GOROOT)
	/tmp/GOPATH/src/bytes (from $GOPATH)
...
```

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -3,6 +3,40 @@
 We'd love to accept your patches and contributions to this project. There are
 just a few small guidelines you need to follow.
 
+## Setup
+
+If you haven't written Go code on your machine before, first follow the official
+Golang instructions for setting up your environment: https://golang.org/doc/code.html
+
+Once you have your environment set up, create a fork of the container-diff repository
+with your personal GitHub account. Then, clone the fork into your `$GOPATH`:
+
+```bash
+git clone git@github.com:<your_account>/container-diff.git 
+$GOPATH/src/github.com/GoogleCloudPlatform &&
+cd $GOPATH/src/github.com/GoogleCloudPlatform/container-diff &&
+git remote add upstream git@github.com:GoogleCloudPlatform/container-diff.git
+```
+
+The last command here sets the official repository as an upstream repository for
+your fork, so you can keep your fork in sync with `MASTER`:
+
+```bash
+(container-diff) git pull upstream master && git push origin master
+```
+
+## Building
+
+From the project root, run `make clean && make`. Under the hood, this essentially
+just runs a `go build` on the project root to create an executable in `/out`.
+
+## Code reviews
+
+All submissions, including submissions by project members, require review. We
+use GitHub pull requests for this purpose. Consult
+[GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
+information on using pull requests.
+
 ## Contributor License Agreement
 
 Contributions to this project must be accompanied by a Contributor License
@@ -14,13 +48,6 @@ your current agreements on file or to sign a new one.
 You generally only need to submit a CLA once, so if you've already submitted one
 (even if it was for a different project), you probably don't need to do it
 again.
-
-## Code reviews
-
-All submissions, including submissions by project members, require review. We
-use GitHub pull requests for this purpose. Consult
-[GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
-information on using pull requests.
 
 ## Tests
 

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -27,8 +27,7 @@ your fork, so you can keep your fork in sync with `MASTER`:
 
 ## Building
 
-From the project root, run `make clean && make`. Under the hood, this essentially
-just runs a `go build` on the project root to create an executable in `/out`.
+From the project root, run `make clean && make`.
 
 ## Code reviews
 

